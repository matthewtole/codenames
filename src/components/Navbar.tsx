import * as React from 'react';
import * as cx from 'classnames';
import { Link } from 'react-router-dom';

interface Props {
  page: 'home' | 'create' | 'join' | 'help';
}

export const Navbar = (props: Props) => {
  return (
    <nav className="navbar is-fixed-top is-dark">
      <div className="navbar-brand">
        <Link className="navbar-item" to="/">
          <img src="/logo.png" width={32} height={32} className="navbar-logo" />
          <strong>Codenames</strong>
        </Link>
        <span className="navbar-burger burger" data-target="navbarMenu">
          <span />
          <span />
          <span />
        </span>
      </div>
      <div id="navbarMenu" className="navbar-menu">
        <div className="navbar-end">
          <Link
            to="/"
            className={cx({
              'navbar-item': true,
              'is-active': props.page === 'home',
            })}
          >
            Home
          </Link>
          <Link
            to="/create"
            className={cx({
              'navbar-item': true,
              'is-active': props.page === 'create',
            })}
          >
            Create Game
          </Link>
          <Link
            to="/join"
            className={cx({
              'navbar-item': true,
              'is-active': props.page === 'join',
            })}
          >
            Join Game
          </Link>
          <Link
            to="/help"
            className={cx({
              'navbar-item': true,
              'is-active': props.page === 'help',
            })}
          >
            Help
          </Link>
          <a
            href="https://github.com/matthewtole/codenames"
            className="navbar-item"
          >
            <i className="fab fa-github" />
          </a>
        </div>
      </div>
    </nav>
  );
};
