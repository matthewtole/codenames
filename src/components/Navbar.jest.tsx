import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Navbar, NavbarProps } from './Navbar';
import { MemoryRouter } from 'react-router';

describe('<Navbar />', () => {
  const defaultProps: NavbarProps = {
    page: 'home',
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navbar {...defaultProps} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
