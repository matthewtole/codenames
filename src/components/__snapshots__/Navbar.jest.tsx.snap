// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<nav
  className="navbar is-fixed-top is-dark"
>
  <div
    className="navbar-brand"
  >
    <a
      className="navbar-item"
      href="/"
      onClick={[Function]}
    >
      <img
        className="navbar-logo"
        height={32}
        src="/logo.svg"
        width={32}
      />
    </a>
    <span
      className="navbar-burger burger"
      data-target="navbarMenu"
      onClick={[Function]}
    >
      <span />
      <span />
      <span />
    </span>
  </div>
  <div
    className="navbar-menu"
    id="navbarMenu"
  >
    <div
      className="navbar-end"
    >
      <a
        className="navbar-item is-active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="navbar-item"
        href="/create"
        onClick={[Function]}
      >
        Create Game
      </a>
      <a
        className="navbar-item"
        href="/join"
        onClick={[Function]}
      >
        Join Game
      </a>
      <a
        className="navbar-item"
        href="/help"
        onClick={[Function]}
      >
        Help
      </a>
      <a
        className="navbar-item"
        href="https://github.com/matthewtole/codenames"
        rel="nofollow"
        target="_blank"
      >
        <i
          className="fab fa-github"
        />
        <span
          className="is-hidden-desktop"
        >
           View on Github
        </span>
      </a>
    </div>
  </div>
</nav>
`;
